spectrum_io/search_result/xisearch.py

```diff
@@ -0,0 +1,231 @@
+                lambda x: any(
+                    mod in str(x).split(";") if pd.notnull(x) else mod is pd.NA for mod in valid_modifications
+                )
+            )
+        ]
+        df = df[
+            df["mods_p2"].apply(
+                lambda x: any(
+                    mod in str(x).split(";") if pd.notnull(x) else mod is pd.NA for mod in valid_modifications
+                )
+            )
+        ]
+
+        return df
+
+    @staticmethod
+    def apply_modifications(split_seq: list, mods: str, mod_positions: str):
+        """
+        Apply modifications to the peptide sequence.
+
+        :param split_seq: List containing the sequence characters
+        :param mods: String containing modifications
+        :param mod_positions: String containing positions of modifications
+        """
+        mod_positions = str(mod_positions)
+
+        if mod_positions in ["nan", "null"]:
+            return
+
+        split_mod = mods.split(";")
+        for idx, mod in enumerate(split_mod):
+            modification = ""
+            if mod == "ox":
+                modification = "M[UNIMOD:35]"
+            elif mod == "cm":
+                modification = "C[UNIMOD:4]"
+            if modification:
+                try:
+                    pos_mod = int(mod_positions.split(";")[idx])
+                    split_seq[pos_mod - 1] = modification
+                except (IndexError, ValueError):
+                    print(f"Error occurred with mod_positions value: {mod_positions}")
+
+    @staticmethod
+    def add_mod_sequence(
+        seq_a: str,
+        seq_b: str,
+        mod_a: str,
+        mod_b: str,
+        crosslinker_position_a: int,
+        crosslinker_position_b: int,
+        mod_a_positions: Union[str, float],
+        mod_b_positions: Union[str, float],
+    ):
+        """
+        Function adds modification in peptide sequence for xl-prosit.
+
+        :param seq_a: unmodified peptide a
+        :param seq_b: unmodified peptide b
+        :param mod_a: all modifications of pep a
+        :param mod_b: all modifications of pep b
+        :param crosslinker_position_a: crosslinker position of peptide a
+        :param crosslinker_position_b: crosslinker position of peptide b
+        :param mod_a_positions: position of all modifications of peptide a
+        :param mod_b_positions: position of all modifications of peptide b
+        :return: modified sequence a and b
+        """
+        mod_a_positions = str(mod_a_positions)  # Ensure it's a string
+        mod_b_positions = str(mod_b_positions)  # Ensure it's a string
+
+        split_seq_a = [x for x in seq_a]
+        split_seq_b = [x for x in seq_b]
+
+        Xisearch.apply_modifications(split_seq_a, mod_a, mod_a_positions)
+        Xisearch.apply_modifications(split_seq_b, mod_b, mod_b_positions)
+
+        split_seq_a[int(crosslinker_position_a) - 1] = "K[UNIMOD:1896]"
+        split_seq_b[int(crosslinker_position_b) - 1] = "K[UNIMOD:1896]"
```

Comment:
do this differently, two crosslinkers are curerntly supported by XL_Prosit, so in the very beginning, the XIsearch object should be called with an additional xl argument storing it as a property in this class. This is then propagated down to this function.